src/class-parsely.php

```diff
@@ -586,4 +544,62 @@ public function get_all_track_types(): array {
 	public function get_default_options() {
 		return $this->option_defaults;
 	}
+
+	/**
+	 * Returns the credentials that are being managed at the platform level.
+	 *
+	 * @since 3.9.0
+	 * @access private
+	 *
+	 * @return Parsely_Options|array<empty> The managed credentials.
+	 */
+	private function get_managed_credentials() {
+		if ( true !== $this->are_credentials_managed ) {
+			return array();
+		}
+
+		$credentials = apply_filters( 'wp_parsely_credentials', false );
+		$result      = array();
+
+		if ( isset( $credentials['site_id'] ) ) {
+			$result['apikey'] = $credentials['site_id'];
+		}
+
+		if ( isset( $credentials['api_secret'] ) ) {
+			$result['api_secret'] = $credentials['api_secret'];
+		}
+
+		if ( isset( $credentials['metadata_secret'] ) ) {
+			$result['metadata_secret'] = $credentials['metadata_secret'];
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Returns whether credentials are being managed at the platform level.
+	 *
+	 * @since 3.9.0
+	 * @access private
+	 *
+	 * @return bool Whether credentials are being managed at the platform level.
+	 */
+	private function are_credentials_managed(): bool {
+		$credentials = apply_filters( 'wp_parsely_credentials', false );
+
+		if ( ! is_array( $credentials ) ) {
+			return false;
+		}
+
+		$required_credentials = array( 'site_id', 'api_secret' );
+		$is_valid             = array();
```

Comment:
The current implementation is fine but we can also remove this `is_valid` array and makes the code a bit simple

```php
private function are_credentials_managed(): bool {
	$credentials = apply_filters( 'wp_parsely_credentials', false );

	if ( ! is_array( $credentials ) ) {
		return false;
	}

	$required_credentials = array( 'site_id', 'api_secret' );

	foreach ( $required_credentials as $key ) {
		if (
			! isset( $credentials[ $key ] ) ||
			! is_string( $credentials[ $key ] ) ||
			'' === $credentials[ $key ]
		) {
			return false;
		}
	}

	return true;
}
```

Reply:
OK, I've updated this in https://github.com/Parsely/wp-parsely/pull/1574/commits/a7dbe1e172e203d474931de98e75beb00e61b331